BackgammonPlay.py

```diff
@@ -0,0 +1,256 @@
+
+    def get_board(self):
+        return self.board
+
+    def start_board(self, sign1, sign2):
+        self.board[0] = 2 * [sign1]
+        self.board[5] = 5 * [sign2]
+        self.board[7] = 3 * [sign2]
+        self.board[11] = 5 * [sign1]
+        self.board[12] = 5 * [sign2]
+        self.board[16] = 3 * [sign1]
+        self.board[18] = 5 * [sign1]
+        self.board[23] = 2 * [sign2]
+
+    def change_board(self, index_now, steps, sign):
+        try:
+            if (sign in self.board[index_now] and (
+                    self.board[index_now + steps] == [] or sign in (self.board[index_now + steps]))):
+                self.board[index_now] = self.board[index_now][:-1]
+                self.board[index_now + steps].append(sign)
+                return
+            else:
+                return InvalidMoveError('invalid move')
+        except Exception as e:
+            return e
+
+    def change_board_level_2(self, index):
+        # get valid input
+        self.board[index] = self.board[index][:-1]
+
+
+class Player(object):
+    def __init__(self, sign, name):
+        self.sign = sign
+        self.name = name
+
+    def get_sign(self):
+        return self.sign
+
+    def get_name(self):
+        return self.name
+
+
+class Game(object):
+    def __init__(self, player1, player2, board):
+        self.p1 = player1
+        self.p2 = player2
+        self.board = board
+        self.count = -1
+        self.val_cube = (0, 0)
+
+    def player_now(self):
+        self.count = self.count + 1
+        if self.count % 2 == 0:
+            return self.p1
+        return self.p2
+
+    def rolling_the_cubes(self):
+        num1 = random.randint(1, 6)
+        num2 = random.randint(1, 6)
+        self.val_cube = num1, num2
+        return self.val_cube
+
+
+class SpecificField(Game):
+    pass
+
+    def print_board(self):
+        print_str_board(self.board.get_board())
+
+    def turn_in_game(self, player):
+        self.val_cube = self.rolling_the_cubes()
+        self.print_board()
+        print("cubes is: {val}. \n{name} please enter your move".format(name=player.get_name(), val=self.val_cube))
+        self.make_move(player)
+
+    def make_move(self, player):
+        dict_cubes = dict_val_cubes(self.val_cube)
+        while (sum(list(dict_cubes.values())) > 0):
+            step = self.move_in_board(player, dict_cubes)
+            dict_cubes = change_dict_val_cubes(dict_cubes, step)
+            self.print_board()
+
+    def get_move_from_user(self, player, dic_val_cubes):
+        index = self.valid_index(player)
+        step = self.legal_step(dic_val_cubes)
+        return index, step
+
+    def move_in_board(self, player, dic_val_cubes):
+        b = " "
+        while (b is not None):
+            print(b)
+            index, step = self.get_move_from_user(player, dic_val_cubes)
+            b = self.board.change_board(index, step, player.get_sign())
+        return step
+
+    def valid_index(self, player):
```

Comment:
there are 2 kind of validation: validate the input to be **legal input** (only int between 0 and 23) and a **legal pawn** (check if there is a pawn that can move in that index).
legal input is the field responsibility
legal move is the game responsibility